### DISCUSSION TITLE
Is there a way to generate 'par.tcl' without running synthesis?

### DISCUSSION THREAD

**Discussion by idwwwoqq808** [ucb-bar/hammer#722] [Support (Q&A)] [answered]
I'm using the example Makefile from chipyard, under directory "chipyard/vlsi". After running "make syn", I deleted the directory 'par-rundir', including 'par.tcl', 'floorplan.tcl', etc., and then run "make par CONFIG=SmallBoomConfig". But hammer does not re-generate the P&R directory and those tcl scripts . The only way I found so far is to run "make syn" again.

Is there a way to let hammer re-generate those tcl scripts for P&R without running synthesis again?

**Comment by idwwwoqq808** [comment]
I'm trying to manually place the SRAMs of Small BOOM chip. I don't want to re-run synthesis every time I change the SRAM placement in YAML configs. That's why I'm asking this question.

**Comment by m-usama-z** [comment]
I'm no expert, but you could try checking the ingredients and target of the "syn" recipe(s) in the hammer.d file in the build directory, then use "touch" to make target of syn newer than the yaml file after editing yaml file. The targets MIGHT be the json files...

**Comment by m-usama-z** [comment]
Or you could try changing SRAM coordinates in the json file that carries the information of synthesis to par (maybe output of synthesis and input of par), instead of the yaml file which is the input to the whole syn+par process...

**Comment by harrisonliew** [comment, marked as answer]
@m-usama-z has the correct suggestions. Alternatively, in Chipyard, you can also do the following:

`make redo-par HAMMER_EXTRA_ARGS='-p <your edited yml file>'`

This breaks the dependency chain and also appends the argument to the underlying Hammer command with your updated constraints. These redo targets are also present in the generated `hammer.d` Make fragment.

**Comment by idwwwoqq808** [comment]
It works! I think I just forgot to add the "HAMMER_EXTRA_ARGS" arguments.
The "par-inpu.json" remains the same but I do see differences in "floorplan.tcl" after re-placing the SRAMs.
Thank you for your answer.
